packages/core/src/lib/select/__tests__/searchable-select.spec.ts

```diff
@@ -1,282 +1,459 @@
 import userEvent from '@testing-library/user-event';
+import type { ComponentProps } from 'svelte';
 
-describe('SearchableSelect', () => {
-  const options = [
-    'First Option',
-    'Option 2',
-    'C.) Option',
-    'Something Else',
-    'With A Whole Lot Of Parts',
-  ];
+import { SearchableSelect as Subject, InputStates } from '$lib';
 
-  const common = { placeholder: 'Select an option', options };
+const onChange = vi.fn();
+const onFocus = vi.fn();
+const onBlur = vi.fn();
 
-  it('Renders the select input', () => {
-    render(SearchableSelect, common);
+const renderSubject = (props: Partial<ComponentProps<Subject>> = {}) => {
+  return render(Subject, {
+    options: ['hello from', 'the other side'],
+    onChange,
+    onFocus,
+    onBlur,
+    ...props,
+  });
+};
+
+const getResults = (): {
+  search: HTMLElement;
+  button: HTMLElement;
+  list: HTMLElement;
+  options: HTMLElement[];
+} => {
+  const search = screen.getByRole('combobox');
+  const button = screen.getByRole('button');
+  const list = screen.getByRole('listbox');
+  const options = within(list).queryAllByRole('option');
+
+  return { search, button, list, options };
+};
+
+describe('combobox list', () => {
+  it('controls a listbox', () => {
+    renderSubject();
+
+    const { search, button, list } = getResults();
+
+    expect(list).toHaveAttribute('id', expect.any(String));
+    expect(button).toHaveAttribute('aria-controls', list.id);
+    expect(search).toHaveAttribute('aria-controls', list.id);
+    expect(search).toHaveAttribute('aria-autocomplete', 'list');
+  });
+
+  it('has a placeholder', () => {
+    renderSubject({ placeholder: "It's me" });
+
+    const { search } = getResults();
 
-    const select = screen.getByPlaceholderText('Select an option');
+    expect(search).toBe(screen.getByPlaceholderText("It's me"));
+  });
 
-    expect(select).toHaveClass(
-      'h-7.5 w-full grow appearance-none border py-1.5 pl-2 pr-1 text-xs leading-tight outline-none'
-    );
+  it.each([
+    {
+      state: InputStates.WARN,
+      classNames: ['border-warning-bright', 'focus:outline-warning-bright'],
+    },
+    {
+      state: InputStates.ERROR,
+      classNames: ['border-danger-dark', 'focus:outline-danger-dark'],
+    },
+  ])('displays a $state state', ({ state, classNames }) => {
+    renderSubject({ state });
+
+    const { search } = getResults();
+
+    expect(search).toHaveClass(...classNames);
   });
 
-  it('Renders the select as disabled', () => {
-    render(SearchableSelect, {
-      ...common,
-      disabled: true,
-    });
+  it('expands the listbox on focus', async () => {
+    const user = userEvent.setup();
+    renderSubject();
+
+    const { search, button, list } = getResults();
 
-    const select = screen.getByPlaceholderText('Select an option');
+    expect(list).toHaveClass('hidden');
+    expect(search).toHaveAttribute('aria-expanded', 'false');
+    expect(button).toHaveAttribute('aria-expanded', 'false');
```

Comment:
These states are driven by the same logic. Perhaps checking all three is redundant in subsequent tests